graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +340,152 @@ def start_global_operations(self):
+        edge_name = fold_scope_location.fold_path[0][1]
+        full_edge_name = '{}_{}'.format(edge_direction, edge_name)
+
+        edge = self._sql_schema_info.join_descriptors[
+            self._current_classname][full_edge_name]
+
+        # we will use this to make sure that the join inside
+        # the subquery uses one alias of the current table
+        # and that the join outside the subquery uses another alias
+        join_info = (edge,
+                     outer_vertex_table,
+                     inner_vertex_type,
+                     self._current_alias)
+
+        self._fold = SQLFoldObject(inner_vertex_table,
+                                   outer_vertex_table,
+                                   inner_vertex_type,
+                                   outer_vertex_type,
+                                   join_info)
+        # find outputs for this fold in self._all_folded_outputs and add to self._fold_outputs
+        if fold_scope_location.fold_path in self._all_folded_outputs:
+            for fold_output in self._all_folded_outputs[fold_scope_location.fold_path]:
+                if fold_output.field == '_x_count':
+                    raise NotImplementedError(u'_x_count not implemented in SQL')
+
+                intermediate_fold_output_name = self._alias_generator.gen_output_column()
+                self._fold_output_column_aliases[fold_output] = intermediate_fold_output_name
+                # add array aggregated intermediate output to self._fold_outputs
+                self._fold.outputs.append(
+                    sqlalchemy.func.array_agg(
+                        self._fold.inner_vertex_table.c[fold_output.field]
+                    ).label(intermediate_fold_output_name)
+                )
+
+        # the group by field is the unique identifier for the outer vertex
+        self._fold.outputs.append(self._fold.outer_vertex_table.c[edge.from_column])
+        self._fold.group_by.append(self._fold.outer_vertex_table.c[edge.from_column])
+
+    def unfold(self):
+        """Complete the execution of a Fold Block."""
+        # Create the folded select statement, updating the fold count and adding the subquery
+        # to self._aliases
+        # left join to the self._from_clause on self._fold_join_vertex_field
+        edge = self._fold.join_info[0]
+        j = self._fold.outer_vertex_table
+        j = sqlalchemy.join(
+            j,
+            self._fold.inner_vertex_table,
+            onclause=(self._fold.outer_vertex_table.c[edge.from_column] ==
+                      self._fold.inner_vertex_table.c[edge.to_column])
+        )
+        folded_subquery = sqlalchemy.select(
+            self._fold.outputs
+        ).select_from(
+            j
+        ).group_by(
+            *self._fold.group_by
+        ).alias(self._alias_generator.gen_subquery())
+
+        self._aliases[(self._fold_inner_vertex_location.fold_path, None)] = folded_subquery
```

Comment:
I don't think this is necessarily a unique key. Isn't it possible to take the same `fold_path` starting from different base locations?

Reply:
Yeah this is tricky because _fold_inner_vertex_location is too specific, but the fold path is both too vague and not specific enough, because the subquery can be defined by multiple outputs inside the fold, so it doesn't make sense to identify it with only one. Furthermore, so is self._fold_inner_vertex_location.base_location is also too vague. 

How do you think I should resolve this?

Reply:
It seems like the key for `self._aliases` is ill defined. Prior to fold, it was simply a `query_path`, but we also need to accommodate `fold_path`s, which are not necessarily unique. 

In the TODO I left about this, I had proposed the use of a tuple of `(query_path, fold_path)`, where the `query_path` comes from the `base_location` for `FoldScopeLocation` and where `fold_path` is `None` for non folded aliases. Does that give you the specificity/uniqueness you need?

Reply:
Hi, you're right. I was mistaken about the relationship between (query_path, fold_path) tuples and folded output columns. I thought each such tuple uniquely identified just one column in a fold, and therefore would've been too specific in the case that there are multiple outputs. FoldScopeLocations uniquely identify output columns (b/c thet contain base_location, fold_path, AND field) so I must've been confused. 

You're right, that system works.

Reply:
@bojanserafimov see this thread on why the keys in the aliases dict are tuples. I was incorrect when I said we could get away without the second part of the tuple.